x/il/keeper/abci.go

```diff
@@ -44,9 +47,19 @@ func (k Keeper) EndBlocker(ctx sdk.Context) {
 
 	// variable for the original value to check if we need to store the new invalidationID value
 	originalInvalidationID := invalidationID
+	deadline := ctx.BlockTime().Unix() + int64(params.EthTimeoutTimestamp)*int64(time.Second)
```

Comment:
Unless there is something going on here that I don't understand, this isn't the right way to calculate the timeout. The timeout is measured in Eth blocks. You need to get the current eth block from gravity, then add a number of blocks to it from your params. Time should not be added to this value.

Reply:
the outgoing logic call from gravity uses blocks but this deadline here is for the redeem liquidity function on uniswap